flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/table/FileSystemTableSink.java

```diff
@@ -386,6 +389,11 @@ private Path toStagingPath() {
         }
     }
 
+    @VisibleForTesting
+    static String getStagingPathPostfix(boolean constant) {
```

Comment:
You introduce the parameter `constant` to make the test parameterizable. But ideally, you don't want to make production code "to please test code". I feel like this `constant` parameter is not necessary (because the production code only uses `false`). The decision whether you return a fixed suffix (or if you follow the comment above to return the actual path, instead: a fixed path) or the random suffix should live in the test code. WDYT?